DuckDuckGo.Fluent.Plugin/DuckResultFactory.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using static DuckDuckGo.Fluent.Plugin.JsonResult;
+using static DuckDuckGo.Fluent.Plugin.AppFunctions;
+
+namespace DuckDuckGo.Fluent.Plugin
+{
+    public class DuckResultFactory
+    {
+        private readonly DuckDuckGoApiResult _apiResult;
+        private readonly string _searchedText;
+
+        private DuckResultFactory(DuckDuckGoApiResult apiResult, string searchedText)
+        {
+            _apiResult = apiResult;
+            _searchedText = searchedText;
+        }
+
+        public static DuckResultFactory Create(DuckDuckGoApiResult root, string searchedText)
+        {
+            return new DuckResultFactory(root, searchedText);
+        }
+
+        public DuckResult GetAnswers()
+        {
+            if (string.IsNullOrWhiteSpace(_apiResult.Answer)) return null;
+            string resultType = _apiResult.AnswerType ?? "Answer";
+            return CreateDuckResult(_apiResult.Answer, resultType, null, ResultType.Answer, true);
+        }
+
+        public DuckResult GetDefinition()
```

Comment:
All the tree methods - `GetAnswer`, `GetDefinition` and `GetAbstract` can be combined to a single method that accepts `ResultType` as parameter. 
This way you will be able to reduce much duplicate code here and in the search app (since you can iterate over the enum values)

Reply:
All three methods have been combined into one and using enum `ResultType` to decide on the type of result.